### ISSUE TITLE
CMR-graphic; failing instance

### ISSUE THREAD

**Issue by rolfvdhulst** [discopt/cmr#25]
The following matrix is _probably_ graphic.
[neos-662469submat.sparse.txt](https://github.com/discopt/cmr/files/9491068/neos-662469submat.sparse.txt)

Running the row-addition graphicness algorithm on it and after each step checking if the fundamental cycles are indeed the ones we expect in the decomposition seems to validly conclude that it's graphic; unless there is a very subtle bug in my code which somehow for all columns manipulates their fundamental cycles in the graph instead of concluding that a matrix is not graphic, CMR is wrongly concluding that the above matrix is NOT graphic. 

CMR reports the following submatrix of the above matrix as the invalid submatrix:
[neos-662469problem.submat.txt](https://github.com/discopt/cmr/files/9491180/neos-662469problem.submat.txt)

However, then running the graphicness algorithm on the 'invalid' submatrix given by this last file, then concludes, which is not consistent with CMR's claim that this matrix was invalid in the first place; so there is at least some bug here already.
[neos-662469problemInvalidSubmat.sparse.txt](https://github.com/discopt/cmr/files/9491181/neos-662469problemInvalidSubmat.sparse.txt)

**Comment by discopt** [comment]
For me it raises an assertion when compiled in debug mode. Can you confirm that this happens for you, too?
```
cmr-graphic: /home/matthias/code/cmr/master.git/src/cmr/graphic.c:2934: CMR_ERROR determineTypeRigid(Dec *, DEC_NEWCOLUMN *, ReducedComponent *, ReducedMember *, int, int, DEC_EDGE *, int): Assertion `numPathEndNodes == 4' failed.
```

**Comment by discopt** [comment]
This is fixed by 3c3d75543a19671fa0ca5e688882dfb36813d9f7. @rolfvdhulst  Can you confirm?
